packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts

```diff
@@ -1,190 +1,147 @@
-      type: 'list',
-      choices: resources,
-    },
-    {
-      name: 'operation',
-      message: 'What would you like to do',
-      type: 'list',
-      when: context.input.command !== 'add',
-      choices: [
-        { name: 'Add another path', value: 'add' },
-        { name: 'Update path', value: 'update' },
-        { name: 'Remove path', value: 'remove' },
-      ],
-    },
-  ];
-
-  const updateApi = await inquirer.prompt(question);
+  const selectedApiName = await prompter.pick<'one', string>('Select the REST API you want to update:', resources);
+  let updateApiOperation = await prompter.pick<'one', string>('What would you like to do?', [
+    { name: 'Add another path', value: 'add' },
+    { name: 'Update path', value: 'update' },
+    { name: 'Remove path', value: 'remove' },
+  ]);
 
   // Inquirer does not currently support combining 'when' and 'default', so
   // manually set the operation if the user ended up here via amplify api add.
   if (context.input.command === 'add') {
-    updateApi.operation = 'add';
+    updateApiOperation = 'add';
   }
 
-  if (updateApi.resourceName === 'AdminQueries') {
+  if (selectedApiName === ADMIN_QUERIES_NAME) {
     const errMessage = `The Admin Queries API is maintained through the Auth category and should be updated using 'amplify update auth' command`;
-    context.print.warning(errMessage);
+    printer.warn(errMessage);
     await context.usageData.emitError(new ResourceDoesNotExistError(errMessage));
     exitOnNextTick(0);
   }
 
-  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
-  const resourceDirPath = path.join(projectBackendDirPath, category, updateApi.resourceName as string);
-  const parametersFilePath = path.join(resourceDirPath, parametersFileName);
-  let parameters;
-  try {
-    parameters = context.amplify.readJsonFile(parametersFilePath);
-  } catch (e) {
-    parameters = {};
+  const projRoot = pathManager.findProjectRoot();
+  if (!stateManager.resourceInputsJsonExists(projRoot, category, selectedApiName)) {
+    // Not yet migrated
+    await migrate(context, projRoot, selectedApiName);
   }
-  parameters.resourceName = updateApi.resourceName;
+
+  const parameters = stateManager.getResourceInputsJson(projRoot, category, selectedApiName);
+  parameters.resourceName = selectedApiName;
 
   Object.assign(allDefaultValues, parameters);
   answers = { ...answers, ...parameters };
   [answers.uuid] = uuid().split('-');
-  const pathList = answers.paths.map(path => path.name);
+  const pathNames = Object.keys(answers.paths);
   let updatedResult = {};
 
-  switch (updateApi.operation) {
+  switch (updateApiOperation) {
     case 'add': {
       updatedResult = pathFlow(context, answers);
       break;
     }
     case 'remove': {
       const pathToRemove = await inquirer.prompt({
         name: 'path',
-        message: 'Please select the path you would want to remove',
+        message: 'Select the path you would want to remove',
```

Comment:
```suggestion
        message: 'Select the path to remove',
```

Reply:
This is a good suggestion, but I think we should address this in a later PR to avoid breaking tests.